src/context/products/product-context.js

```diff
@@ -0,0 +1,44 @@
+import { createContext, useReducer, useContext } from "react";
+import { productReducerFunc } from "../../reducer/product-reducer";
+import { products } from "../../backend/db/products";
+import { getSortedList } from "../../utils";
+
+const productContext = createContext();
+
+const ProductProvider = ({ children }) => {
+  const initialState = {
+    sortBy: null,
+    categories: [],
+    priceRange: 2500,
+    ratings: 0,
+  };
+
+  const getFilteredProducts = (state, sortedList) => {
+    const filterList = sortedList
+      .filter((item) =>
+        state.categories.length === 0
+          ? item
+          : state["categories"].includes(item.categoryName)
```

Comment:
you could have used state.categories.icludes here, is there any specific reason for using this syntax, bcz as per my understanding this is normally used when the key is unknown, but here you know the key as you have also used "" in state["categories"]. Please correct me if i m wrong!!
Rest of the code is good!!

Reply:
You are right, can use either of that
I will change to maintain consistency
Thank you